### PR TITLE
Add MONAI loss configuration classes

### LINKED ISSUE #907: Config class for loss functions from MONAI
PyTorch's loss functions are quite limited. Even if MONAI's loss function are supported currently, they don't have associated config classes, which limits reproducibility.

### PR DESCRIPTION
**Pull request by txmed82**
Closes #907

## Summary

- add typed, serializable configs for Dice, DiceCE, DiceFocal, generalized Dice, generalized Dice-Focal, Focal, and Tversky MONAI losses
- register the configs with the existing loss enum and factory while preserving MONAI 1.5.0 defaults
- validate activation exclusivity and bounded or non-negative parameters, with support for scalar and list class weights
- document the new public configuration classes

## Why

Raw MONAI losses already work in ClinicaDL, but their constructor parameters are not preserved in model configuration JSON. These config classes make common segmentation-loss choices reproducible and reloadable.

## Validation

- `pytest tests/unittests/losses -q` — 78 passed
- `pytest tests/unittests/models/test_factory.py tests/unittests/models/test_supervised.py tests/unittests/models/test_reconstruction.py tests/unittests/metrics/test_loss.py -q` — 5 passed
- affected pre-commit hooks — passed
- `make -C docs html` — succeeded with 7 pre-existing warnings
- a broader CPU unit run reached 959 passed, 1 skipped, and 12 deselected before being stopped during slow training tests; two environment-only failures caused by missing `pip` were rerun after installing it, and both passed

### CODE REVIEW

**Comment by thibaultdvx** [comment]
Hey @txmed82, thanks for the PR! I will have a look at it tomorrow

File: tests/unittests/losses/test_factory.py [hunk 1]
```diff
@@ -30,3 +37,15 @@ def test_get_loss_function_from_dict(config, tmp_path):
...
+
+@pytest.mark.parametrize(
+    "config",
+    [
+        DiceLossConfig(sigmoid=True, weight=[1, 2]),
+        GeneralizedDiceLossConfig(w_type="simple", soft_label=True),
+        FocalLossConfig(gamma=3, alpha=0.25, use_softmax=True),
+    ],
+)
+def test_monai_loss_non_default_round_trip(config):
```
Comment (thibaultdvx) [inline tests/unittests/losses/test_factory.py:52]: Could you add a docstring to explain why this test is necessary?

**Review by thibaultdvx — APPROVED** [review]
LGTM! Thanks for the great work :)

File: CITATION.cff [hunk 2]
```diff
@@ -37,6 +37,10 @@ authors:
     given-names: Ghislain
   - family-names: Wen
     given-names: Junhao
+  - family-names: Son
+    given-names: Colin
+    website: https://github.com/txmed82
+    orcid: https://orcid.org/0000-0002-1782-0537
```
Comment (thibaultdvx) [inline CITATION.cff:43]: Hey @txmed82, could you please make sure that this information is correct. Feel free to add your `email` or `affiliation‎`.

**Review by thibaultdvx — APPROVED** [review]
